frontend/src/components/admin/AddLanguageModal.tsx

```diff
@@ -0,0 +1,218 @@
+    return { value: "Left-to-right" };
+  };
+
+  return (
+    <>
+      <Modal
+        isOpen={isOpen}
+        motionPreset="slideInBottom"
+        onClose={onClose}
+        size="6xl"
+      >
+        <ModalOverlay />
+        <ModalContent paddingLeft="32px">
+          <ModalCloseButton
+            marginLeft="auto"
+            marginRight="8px"
+            marginTop="8px"
+            position="static"
+          />
+          <ModalHeader paddingTop="20px">
+            <Heading as="h3" size="lg">
+              Add New Language
+            </Heading>
+          </ModalHeader>
+          <ModalBody paddingBottom="36px">
+            <Flex justifyContent="space-between">
+              <Flex direction="column" width="40%" margin="10px">
+                <Heading as="h4" size="sm" marginBottom="3px">
+                  Add New Language
+                </Heading>
+                <Text marginBottom="3px">
+                  Please enter the name of the new language.
+                </Text>
+                <Input
+                  id="new-language"
+                  type="text"
+                  value={language}
+                  onChange={(event) => setLanguage(event.target.value)}
+                  placeholder="Enter new language"
+                  marginBottom="10px"
+                />
+                <Heading as="h4" size="sm" marginBottom="3px">
+                  Script Direction
+                </Heading>
+                <Text marginBottom="3px">
+                  Please enter the direction of the script. For example, English
+                  is left-to-right and Arabic is right-to-left.
+                </Text>
+                <Box
+                  width="80%"
+                  sx={language === "" ? disabledStyle : undefined}
+                >
+                  <Select
+                    placeholder="Select Script Direction"
+                    options={scriptDirectionOptions}
+                    onChange={(option: any) =>
+                      setIsRtl(option.value === "Right-to-left")
+                    }
+                    getOptionLabel={(option: any) => option.value}
+                    value={getScriptDirectionSelectValue()}
+                    styles={colourStyles}
+                    components={{ DropdownIndicator }}
+                    isDisabled={language === ""}
+                  />
+                </Box>
+              </Flex>
+              <Flex direction="column" margin="10px" width="50%">
+                <Heading as="h4" size="sm">
+                  Existing Languages
+                </Heading>
+                <Box overflow="auto" height="300px" padding="10px">
+                  <Table
+                    borderRadius="12px"
+                    boxShadow="0px 0px 2px grey"
+                    margin="auto"
+                    size="sm"
+                    theme="gray"
+                    variant="striped"
+                    width="100%"
+                  >
```

Comment:
Nit: the existing languages table scrolls a bit strangely, maybe try something like:
```
<Box
  borderRadius="10px"
  boxShadow="0px 0px 2px grey"
  height="300px"
  overflow="auto"
>
  <Table
    marginTop="-10px"
    size="sm"
    theme="gray"
    variant="striped"
    width="100%"
  >
```